test/formal/checked_int_add.py

```diff
@@ -0,0 +1,39 @@
+from rule import Rule
+from opcodes import *
+from util import *
+
+"""
+Overflow checked signed integer addition.
+"""
+
+n_bits = 256
+type_bits = 8
+
+while type_bits <= n_bits:
+
+	rule = Rule()
+
+	# Input vars
+	X_short = BitVec('X', type_bits)
+	Y_short = BitVec('Y', type_bits)
+
+	# Z3's overflow and underflow conditions
+	actual_overflow = (BVAddNoOverflow(X_short, Y_short, True) == False)
+	actual_underflow = (BVAddNoUnderflow(X_short, Y_short) == False)
+
+	# cast to full n_bits values
+	X = BVSignedUpCast(X_short, n_bits)
+	Y = BVSignedUpCast(Y_short, n_bits)
+
+	# Constants
+	maxValue = BVSignedMax(type_bits, n_bits)
+	minValue = BVSignedMin(type_bits, n_bits)
+
+	# Overflow and underflow checks in YulUtilFunction::overflowCheckedIntAddFunction
+	overflow_check = AND(ISZERO(SLT(X, 0)), SGT(Y, SUB(maxValue, X)))
+	underflow_check = AND(SLT(X, 0), SLT(Y, SUB(minValue, X)))
```

Comment:
Nice, these look exactly like the yul code

Reply:
Yeah, almost... I'm not sure, if a lower-case version of the builtins would have naming collisisions with anything - that'd be perfect... probably good enough, though.